### PR TITLE
feat: add meta data

### LINKED ISSUE #728: [Feature]: [SEO] add meta tags
### How the project can be improved?

VSF integration should add the defaults SEO meta tags for at least the product and categories pages.
At least we should see the meta **title** and meta **description** coming up from the Magento data.

### What are the acceptance criteria?

- [ ]

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by haelbichalex**
## Description

* Add title and meta tags for product
* Add meta_keywords to cms page

## Related Issue

Closes #728.

## Motivation and Context

Merchants can use the provided meta fields in Magento for products and cms pages.

## How Has This Been Tested?

Add data to the **Search Engine Optimization** fields in Magento for a product and a cms page and check, if these values are applied to the `head` tag.

## Screenshots (if appropriate):
meta_tags

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Frodigo** [comment]
Hi @haelbichalex could we help you somehow with this?

**Comment by jonathanribas** [comment]
Hi @haelbichalex, as far as I remember, meta's are also missing on category pages :( 
Any chances to add them?

**Comment by jonathanribas** [comment]
Hi @Frodigo, any update on this one please?

**Comment by Frodigo** [comment]
HI @haelbichalex @jonathanribas in terms of category pages, this is blocked for now until we refactor the deprecated UrlResolver query to the route GraphQL query.

For CMS pages and product page, we can proceed with this PR so please resolve conflicts and finish it for product and CMS if you have a chance. 

When we finish refactor, it will be possible tog get meta tags for category as well.

**Comment by haelbichalex** [comment]
@Frodigo Merged latest `develop`, so you can review this now IMO.

File: packages/theme/modules/catalog/pages/product.vue [hunk 1]
```diff
@@ -603,6 +603,30 @@ export default defineComponent({
...
+        name: 'keywords',
+        content: this.product.meta_keyword,
+      });
+    }
+
+    return {
+      title: this.product.meta_title,
+      meta,
+    };
+  },
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/pages/product.vue:629]: We are using the same logic for building meta data in multiple places, like cms pages, categories, product, maybe more. Could add a separate script responsible for building metadata that can be used in multiple places? It should wrap repeated logic.

File: packages/theme/pages/Page.vue [hunk 2]
```diff
@@ -59,7 +59,7 @@ export default defineComponent({
         return nuxtError({ statusCode: 404, message: app.i18n.t('Page not found') });
       }
 
-      addTags([{ prefix: CacheTagPrefix.View, value: page.value.identifier }]);
+      return addTags([{ prefix: CacheTagPrefix.View, value: page.value.identifier }]);
```
Comment (bartoszherba) [inline packages/theme/pages/Page.vue:62]:
```suggestion
     addTags([{ prefix: CacheTagPrefix.View, value: page.value.identifier }]);
```

Comment (bartoszherba) [reply packages/theme/pages/Page.vue:62]: + return true;
 add Tags returns void

**Comment by Frodigo** [comment]
Closing because of inactivity, feel free to reopen when you want to continue work on this.

**Comment by jonathanribas** [comment]
> HI @haelbichalex @jonathanribas in terms of category pages, this is blocked for now until we refactor the deprecated UrlResolver query to the route GraphQL query.
> 
> For CMS pages and product page, we can proceed with this PR so please resolve conflicts and finish it for product and CMS if you have a chance.
> 
> When we finish refactoring, it will be possible to get meta tags for category as well.

Hi @Frodigo, do you have news regarding the GraphQL get for a category refactoring?

**Comment by Frodigo** [comment]
Hi @jonathanribas, yep, refactoring is done. If you have a chance, please take a look ikt again. 

It would be great if you add changes proposed by @bartoszherba, resolve build issues, and then we can proceed with this.

**Comment by bartoszherba** [comment]
Closed due to a long inactivity.

**Pull request closed without merging**